### PR TITLE
Refactors and fixes summon events/guns/magic

### LINKED ISSUE #64967: Summon Guns / Summon Magic runtime to hell when purchased by a wizard
## Round ID:

[178532](https://scrubby.melonmesa.com/round/178532) 

## Reproduction:

- Have a 100 threat round.
- Have a Wizard buy Summon Guns and Summon Magic.
- No one gets Magic or Guns, because there are now like 200 runtimes from both.
  - (Trigger the event manually because they got cucked)

### PR DESCRIPTION
**Pull request by MrMelbert**
## About The Pull Request

Closes #64967

- Refactors the summon guns and summon events controllers into one datum to reduce code copy+paste.
- Removed the weird duplicate use of the proc `rightandwrong` which didn't really server a purpose to be one proc. Split it into `summon_guns()` and `summon_magic()`. 
- Reduced copy+paste in ritual purchasing code. 
- Fixes the inability to purchase summon events multiple times. 

## Why It's Good For The Game

One in a million chance of getting 100 threat and they're both broken.
A lot of the cod was copy+pasted around and could be condensed. Win-win?

## Changelog

:cl: Melbert
fix: Fixes summon magic and summon guns
refactor: Refactored wizard rituals (events/ghosts/guns/magic/madness) to reduce code copy+paste.
/:cl:

### CODE REVIEW

File: code/modules/antagonists/wizard/equipment/spellbook.dm [hunk 1]
```diff
@@ -539,102 +539,105 @@
 	desc = "A \"Friendly\" Wizard will protect the station, and try to kill you. They get a spellbook much like you, but will use it for \"GOOD\"."
 
 /// How much threat we need to let these rituals happen on dynamic
-#define MINIMUM_THREAT_FOR_RITUALS 100
+#define MINIMUM_THREAT_FOR_RITUALS 0 //100
```
Comment (cacogen) [inline code/modules/antagonists/wizard/equipment/spellbook.dm:542]: keep this in

Comment (MrMelbert) [reply code/modules/antagonists/wizard/equipment/spellbook.dm]: that was just for testing lol

**Comment by timothymtorres** [comment]
Refactors don't get the same kind of respect as features or fixes, but this is really nice!  Magic code right now is atrocious and really needs a large overhaul.  Lots of things c/p'd and poorly named variables.

File: code/modules/antagonists/wizard/equipment/spellbook.dm [hunk 2]
```diff
@@ -544,97 +544,85 @@
...
 /datum/spellbook_entry/summon/guns
 	name = "Summon Guns"
 	desc = "Nothing could possibly go wrong with arming a crew of lunatics just itching for an excuse to kill you. There is a good chance that they will shoot each other first."
 
 /datum/spellbook_entry/summon/guns/IsAvailable()
-	if(!SSticker.mode) // In case spellbook is placed on map
-		return FALSE
-	if(istype(SSticker.mode, /datum/game_mode/dynamic)) // Disable events on dynamic
+	// Summon Guns requires 100 threat.
+	if(istype(SSticker.mode, /datum/game_mode/dynamic))
```
Comment (Mothblocks) [inline code/modules/antagonists/wizard/equipment/spellbook.dm:573]: Remove this check entirely, Dynamic will forever be the last gamemode, I was very close to killing it but got proeccupied.

File: code/modules/antagonists/wizard/equipment/spellbook.dm [hunk 3]
```diff
@@ -544,97 +544,85 @@
...
 /datum/spellbook_entry/summon/magic
 	name = "Summon Magic"
 	desc = "Share the wonders of magic with the crew and show them why they aren't to be trusted with it at the same time."
 
 /datum/spellbook_entry/summon/magic/IsAvailable()
-	if(!SSticker.mode) // In case spellbook is placed on map
-		return FALSE
-	if(istype(SSticker.mode, /datum/game_mode/dynamic)) // Disable events on dynamic
+	// Summon Magic requires 100 threat.
+	if(istype(SSticker.mode, /datum/game_mode/dynamic))
```
Comment (Mothblocks) [inline code/modules/antagonists/wizard/equipment/spellbook.dm:591]: Same here

File: code/modules/antagonists/wizard/equipment/spellbook.dm [hunk 4]
```diff
@@ -544,97 +544,85 @@
...
 	cost = 2
-	limit = 1
+	limit = 5 // Each purchase can intensify it.
 
 /datum/spellbook_entry/summon/events/IsAvailable()
-	if(!SSticker.mode) // In case spellbook is placed on map
-		return FALSE
-	if(istype(SSticker.mode, /datum/game_mode/dynamic)) // Disable events on dynamic
+	// Summon Events requires 100 threat.
+	if(istype(SSticker.mode, /datum/game_mode/dynamic))
```
Comment (Mothblocks) [inline code/modules/antagonists/wizard/equipment/spellbook.dm:611]: Same here

File: code/modules/antagonists/wizard/equipment/spellbook.dm [hunk 5]
```diff
@@ -544,97 +544,85 @@
...
 	playsound(get_turf(user), 'sound/magic/castsummon.ogg', 50, TRUE)
-	to_chat(user, span_notice("You have cast summon events."))
-	return TRUE
+	return ..()
 
 /datum/spellbook_entry/summon/events/GetInfo()
-	if(times>0)
-		. += "You cast it [times] times.<br>"
+	if(times > 0)
+		. += "You have cast it [times] time[times == 1 ? "":"s"].<br>"
```
Comment (Mothblocks) [inline code/modules/antagonists/wizard/equipment/spellbook.dm:625]:
```suggestion
		. += "You have cast it [times] time\s.<br>"
```

**Review by Mothblocks — CHANGES_REQUESTED** [review]

**Pull request merged**